fractor-xml/src/AbstractXmlFractor.php

```diff
@@ -7,7 +7,7 @@
 
 abstract class AbstractXmlFractor implements DomNodeVisitor, XmlFractor
 {
-    public function beforeTraversal(\DOMNode $rootNode): void
+    public function beforeTraversal(\DOMDocument $rootNode): void
```

Comment:
A DOMDocument is passed to the method and if you declare it in the same way like I do with a local variable, you get a mismatch of parameters. This unifies it and specifies what kind of dom node you get.

Reply:
Sounds logical. I was opposed to this change at first, but since we will very likely always iterate complete documents, (the name `DomDocumentIterator` also hints at that fact actually…), this change is fine IMO.